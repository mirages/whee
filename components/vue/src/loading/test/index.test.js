import Loading from '..'
import { mount } from '@vue/test-utils'

test('the size prop can set the loading size', () => {
  const wrapper = mount(Loading, {
    props: {
      size: 40
    }
  })

  const loadingSize = wrapper.get('[data-test=loading-size]').element

  expect(loadingSize.style.width).toEqual('40px')
  expect(loadingSize.style.height).toEqual('40px')
  expect(wrapper.html()).toMatchSnapshot()
})

test('the type prop can set the loading type: spinner or circle', async () => {
  const wrapper = mount(Loading, {
    props: {
      type: 'spinner'
    }
  })

  expect(wrapper.find('[data-test=loading-spinner]').exists()).toBe(true)
  let prevHtml = wrapper.html()

  await wrapper.setProps({ type: 'circle' })
  expect(wrapper.find('[data-test=loading-circle]').exists()).toBe(true)
  expect(prevHtml + wrapper.html()).toMatchSnapshot()
})

test('the default slot can set the loading text and style', () => {
  const wrapper = mount(Loading, {
    slots: {
      default: () => <p class="test">loading...</p>
    }
  })

  expect(wrapper.html()).toContain('<p class="test">loading...</p>')
  expect(wrapper.html()).toMatchSnapshot()
})
