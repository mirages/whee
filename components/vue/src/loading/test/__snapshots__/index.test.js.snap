// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the default slot can set the loading text and style 1`] = `
<div class="loading">
  <div data-test="loading-size"
       class="spinner"
       style="width: 20px; height: 20px;"
  >
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(-90deg); transform-origin: -60% 50%; opacity: 1; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(-60deg); transform-origin: -60% 50%; opacity: 0.9416666666666667; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(-30deg); transform-origin: -60% 50%; opacity: 0.8833333333333333; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(0deg); transform-origin: -60% 50%; opacity: 0.825; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(30deg); transform-origin: -60% 50%; opacity: 0.7666666666666667; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(60deg); transform-origin: -60% 50%; opacity: 0.7083333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(90deg); transform-origin: -60% 50%; opacity: 0.65; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(120deg); transform-origin: -60% 50%; opacity: 0.5916666666666668; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(150deg); transform-origin: -60% 50%; opacity: 0.5333333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(180deg); transform-origin: -60% 50%; opacity: 0.4750000000000001; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(210deg); transform-origin: -60% 50%; opacity: 0.41666666666666674; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(240deg); transform-origin: -60% 50%; opacity: 0.3583333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
  </div>
  <div style="color: rgb(153, 153, 153);">
    <p class="test">
      loading...
    </p>
  </div>
</div>
`;

exports[`the size prop can set the loading size 1`] = `
<div class="loading">
  <div data-test="loading-size"
       class="spinner"
       style="width: 40px; height: 40px;"
  >
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(-90deg); transform-origin: -60% 50%; opacity: 1; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(-60deg); transform-origin: -60% 50%; opacity: 0.9416666666666667; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(-30deg); transform-origin: -60% 50%; opacity: 0.8833333333333333; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(0deg); transform-origin: -60% 50%; opacity: 0.825; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(30deg); transform-origin: -60% 50%; opacity: 0.7666666666666667; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(60deg); transform-origin: -60% 50%; opacity: 0.7083333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(90deg); transform-origin: -60% 50%; opacity: 0.65; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(120deg); transform-origin: -60% 50%; opacity: 0.5916666666666668; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(150deg); transform-origin: -60% 50%; opacity: 0.5333333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(180deg); transform-origin: -60% 50%; opacity: 0.4750000000000001; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(210deg); transform-origin: -60% 50%; opacity: 0.41666666666666674; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 12px; height: 4px; transform: translate(60%, -50%) rotate(240deg); transform-origin: -60% 50%; opacity: 0.3583333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
  </div>
  <div style="color: rgb(153, 153, 153);">
  </div>
</div>
`;

exports[`the type prop can set the loading type: spinner or circle 1`] = `
<div class="loading">
  <div data-test="loading-size"
       class="spinner"
       style="width: 20px; height: 20px;"
  >
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(-90deg); transform-origin: -60% 50%; opacity: 1; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(-60deg); transform-origin: -60% 50%; opacity: 0.9416666666666667; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(-30deg); transform-origin: -60% 50%; opacity: 0.8833333333333333; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(0deg); transform-origin: -60% 50%; opacity: 0.825; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(30deg); transform-origin: -60% 50%; opacity: 0.7666666666666667; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(60deg); transform-origin: -60% 50%; opacity: 0.7083333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(90deg); transform-origin: -60% 50%; opacity: 0.65; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(120deg); transform-origin: -60% 50%; opacity: 0.5916666666666668; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(150deg); transform-origin: -60% 50%; opacity: 0.5333333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(180deg); transform-origin: -60% 50%; opacity: 0.4750000000000001; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(210deg); transform-origin: -60% 50%; opacity: 0.41666666666666674; background-color: rgb(153, 153, 153);"
    >
    </i>
    <i data-test="loading-spinner"
       style="width: 6px; height: 2px; transform: translate(60%, -50%) rotate(240deg); transform-origin: -60% 50%; opacity: 0.3583333333333334; background-color: rgb(153, 153, 153);"
    >
    </i>
  </div>
  <div style="color: rgb(153, 153, 153);">
  </div>
</div>
<div class="loading">
  <div data-test="loading-size"
       class="circle"
       style="width: 20px; height: 20px;"
  >
    <svg viewbox="0 0 30 30">
      <circle cx="15"
              cy="15"
              r="12"
              data-test="loading-circle"
              stroke="#999"
      >
      </circle>
    </svg>
  </div>
  <div style="color: rgb(153, 153, 153);">
  </div>
</div>
`;
